src/showcase/app/shared/component-viewer/stackblitz-writer/stackblitz-writer.service.ts

```diff
@@ -0,0 +1,207 @@
+  '@angular/compiler': '*',
+  '@angular/core': '*',
+  '@angular/forms': '*',
+  '@angular/platform-browser': '*',
+  '@angular/platform-browser-dynamic': '*',
+  '@angular/router': '*',
+  '@sbb-esta/angular-core': '*',
+  '@sbb-esta/angular-icons': '*',
+  '@sbb-esta/angular-public': '*',
+  '@sbb-esta/angular-business': '*',
+  'core-js': '2.6.9',
+  rxjs: '*',
+  tslib: '*',
+  'zone.js': '*',
+};
+
+/**
+ * StackBlitz writer, write example files to StackBlitz.
+ *
+ * StackBlitz API
+ * URL: https://run.stackblitz.com/api/aio/v1/
+ * data: {
+ *   // File name, directory and content of files
+ *   files[file-name1]: file-content1,
+ *   files[directory-name/file-name2]: file-content2,
+ *   // Can add multiple tags
+ *   tags[0]: tag-0,
+ *   // Description of StackBlitz
+ *   description: description,
+ *   // Private or not
+ *   private: true
+ *  // Dependencies
+ *  dependencies: dependencies
+ * }
+ */
+@Injectable()
+export class StackblitzWriterService {
+  constructor(private _http: HttpClient) {}
+
+  /**
+   * Returns an HTMLFormElement that will open a new StackBlitz template with the example data when
+   * called with submit().
+   */
+  constructStackBlitzForm(data: ExampleData): Promise<HTMLFormElement> {
+    const indexFile = `${data.indexFilename}.ts`;
+    const form = this._createFormElement(indexFile);
+
+    TAGS.forEach((tag, i) => this._appendFormInput(form, `tags[${i}]`, tag));
+    this._appendFormInput(form, 'private', 'true');
+    this._appendFormInput(form, 'description', data.description);
+    this._appendFormInput(form, 'dependencies', JSON.stringify(dependencies));
+
+    return new Promise((resolve) => {
+      this._addFileToForm(
+        form,
+        data,
+        `<sbb-${data.selectorName}>loading</sbb-${data.selectorName}>`,
+        'src/index.html',
+        '',
+        false
+      );
+
+      TEMPLATE_FILES.forEach((file) =>
+        this._readFile(file, TEMPLATE_PATH).subscribe(
+          (response: string) => {
+            if (file.includes('app.module.ts')) {
+              // add currently selected component to declarations and bootstrap it
+              response = response.replace(/\{0\}/g, data.componentName);
+              response = response.replace(/\{1\}/g, data.selectorName);
+            } else if (file.includes('sbb.module.ts')) {
+              // import public or business module
+              response = response.replace(
+                /\{0\}/g,
+                data.business ? '@sbb-esta/angular-business' : '@sbb-esta/angular-public'
+              );
+            }
+            this._addFileToForm(form, data, response, file, TEMPLATE_PATH, false);
+          },
+          (error) => console.log(error)
+        )
+      );
+
+      // remove ".component" filename section from html template URL
+      data.exampleFiles.some((file) => {
+        if (file.name.includes('.ts')) {
+          file.content = file.content.replace('component.html', 'html');
```

Comment:
Since we compile scss with bazel, we have .css endings in styleUrl, but the source files end in .scss
Also replace `.css` with `.scss`